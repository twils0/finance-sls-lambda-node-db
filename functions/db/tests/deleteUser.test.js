const { deleteUser } = require('../deleteUser');

global.pool = {
  query: jest.fn(),
};

const userId = 'testUserID';

describe('functions', () => {
  describe('db', () => {
    describe('deleteUser', () => {
      afterEach(() => {
        global.pool.query.mockReset();
      });

      it('fails and returns unexpected error, when global.pool.query throws error on first query', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.reject(error);
        });

        try {
          await deleteUser(userId);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('fails and returns unexpected error, when global.pool.query throws error on second query', async () => {
        const error = {
          code: 'testCode',
          message: 'testMessage',
        };
        global.pool.query.mockImplementationOnce((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.resolve(null);
        });
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.reject(error);
        });

        try {
          await deleteUser(userId);
        } catch (errorCatch) {
          expect(errorCatch).toEqual(error);
        }
      });

      it('correctly returns user id', async () => {
        global.pool.query.mockImplementation((dbString, valuesArray) => {
          expect(dbString).toMatchSnapshot();
          expect(valuesArray).toEqual([userId]);
          return Promise.resolve(null);
        });

        const result = await deleteUser(userId);

        expect(result).toEqual(null);
      });
    });
  });
});
