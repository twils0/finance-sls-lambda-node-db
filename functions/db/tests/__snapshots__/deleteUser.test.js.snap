// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions db deleteUser correctly returns user id 1`] = `
"DELETE FROM \\"user\\".\\"join_user_security\\"
          WHERE user_id = $1::uuid;"
`;

exports[`functions db deleteUser correctly returns user id 2`] = `
"DELETE FROM \\"user\\".\\"user\\"
        WHERE user_id = $1::uuid;"
`;

exports[`functions db deleteUser fails and returns unexpected error, when global.pool.query throws error on first query 1`] = `
"DELETE FROM \\"user\\".\\"join_user_security\\"
          WHERE user_id = $1::uuid;"
`;

exports[`functions db deleteUser fails and returns unexpected error, when global.pool.query throws error on second query 1`] = `
"DELETE FROM \\"user\\".\\"join_user_security\\"
          WHERE user_id = $1::uuid;"
`;

exports[`functions db deleteUser fails and returns unexpected error, when global.pool.query throws error on second query 2`] = `
"DELETE FROM \\"user\\".\\"user\\"
        WHERE user_id = $1::uuid;"
`;
